randomUsernameAndPassword/assignment1/Q2.py

```diff
@@ -0,0 +1,54 @@
+import unittest
+
+class Node:
+  def __init__(self, values):
+    assert(values)
+    head, tail = values[0], values[1:]
+    self.data = head
+    if tail:
+      self.next = Node(tail)
+    else:
+      self.next = None
+
+  def get_next(self):
+    return self.next
+
+  def get_data(self):
+    return self.data
+
+def get_kth_last(head, k):
+  if (head == None or k < 0):
+    return None
```

Comment:
returning None is uncommon and is ambiguous as the k-th element could actually be None and you would have no way to distinguish the two.

It is much more common to raise an Exception:

>>> [1,2,3][10]
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
IndexError: list index out of range

Can you raise an IndexError?

Reply:
Fixed